lib/active_reporting/reporting_dimension.rb

```diff
@@ -149,70 +166,75 @@ def validate_against_datetime_hierarchies(hierarchical_label)
     end
 
-    def degenerate_fragment
-      return "#{name}_#{@label}" if datetime?
-      "#{model.quoted_table_name}.#{name}"
-    end
-
-    def degenerate_select_fragment
-      if datetime?
-        if model.connection.adapter_name == "Mysql2"
-          return degenerate_select_fragment_mysql
-        else # Postgress
-          return degenerate_select_fragment_postgress
-        end
+    def datetime_drill_label_fragment(select_statement)
+      if model.connection.adapter_name == "Mysql2"
+        datetime_drill_mysql(select_statement)
+      else # Postgress
+        datetime_drill_postgress(select_statement)
       end
-      "#{model.quoted_table_name}.#{name}"
     end
 
-    def degenerate_select_fragment_postgress
-      "DATE_TRUNC('#{@label}', #{model.quoted_table_name}.#{name}) AS #{name}_#{@label}"
+    def datetime_drill_postgress(select_statement)
+      "DATE_TRUNC('#{@datetime_drill}', #{select_statement})"
     end
 
-    def degenerate_select_fragment_mysql
-      field = "#{model.quoted_table_name}.#{model.connection.quote_column_name(name)}"
-      modifier = case @label.to_sym
+    def datetime_drill_mysql(select_statement)
+      case @datetime_drill.to_sym
       when :microseconds
-        "MICROSECOND(#{field})"
+        "MICROSECOND(#{select_statement})"
       when :milliseconds
-        "MICROSECOND(#{field}) DIV 1000"
+        "MICROSECOND(#{select_statement}) DIV 1000"
       when :second
-        "SECOND(#{field})"
+        "SECOND(#{select_statement})"
       when :minute
-        "MINUTE(#{field})"
+        "MINUTE(#{select_statement})"
       when :hour
-        "HOUR(#{field})"
+        "HOUR(#{select_statement})"
       when :day
-        "DAY(#{field})"
+        "DAY(#{select_statement})"
       when :week
-        "WEEKDAY(#{field})"
+        "WEEKDAY(#{select_statement})"
       when :month
-        "MONTH(#{field})"
+        "MONTH(#{select_statement})"
       when :quarter
-        "QUARTER(#{field})"
+        "QUARTER(#{select_statement})"
       when :year
-        "YEAR(#{field})"
+        "YEAR(#{select_statement})"
       when :decade
-        "YEAR(#{field}) DIV 10"
+        "YEAR(#{select_statement}) DIV 10"
       when :century
-        "YEAR(#{field}) DIV 100"
+        "YEAR(#{select_statement}) DIV 100"
       when :millennium
-        "YEAR(#{field}) DIV 1000"
+        "YEAR(#{select_statement}) DIV 1000"
       end
-      "#{modifier} AS #{name}_#{@label}"
     end
 
     def identifier_fragment
       "#{klass.quoted_table_name}.#{model.connection.quote_column_name(klass.primary_key)}"
     end
 
+    def identifier_fragment_alias
+      "#{model.connection.quote_column_name("#{name}_identifier")}"
+    end
+
     def label_fragment
-      "#{klass.quoted_table_name}.#{model.connection.quote_column_name(@label)}"
+      ss = "#{klass.quoted_table_name}.#{model.connection.quote_column_name(@label)}"
```

Comment:
Similar to a previous comment, I presume `ss` is shorthand for "select_statement" which is not what this is. It's the label fragment. Perhaps a local variable named `fragment` would make the method more readable?

Reply:
Done!